### PR TITLE
Adds definition signature to glossary

### LINKED ISSUE #704: Add "signature" to glossary
We use "signed" and "signature" through the docs, including within the glossary itself.

### PR DESCRIPTION
**Pull request by s-ben**
Took a crack at adding a definition for the signature (below) ( Closes #704 ).

#### Signature

A cryptographic mechanism that proves that the owner of the private key has approved a transaction.

### CODE REVIEW

File: docs/glossary.md [hunk 1]
```diff
@@ -270,6 +270,10 @@ A proposal to change the consensus rules of the Decred blockchain. Rule change p
 
 A 256-bit (32-byte) cryptographically secure random number that can be used to recover a wallet. When creating a wallet using any Decred wallet software, a 33 word "seed phrase" is also created from the 32-byte seed. If the wallet is lost or corrupted, the seed phrase can be used to restore the wallet's private keys, transaction history, and balances using any Decred wallet on any computer.
 
+#### Signature
+
+A cryptographic mechanism that proves that the owner of the private key has approved a transaction.
```
Comment (davecgh) [inline docs/glossary.md]: Signatures are used for more than transactions.

Comment (s-ben) [reply docs/glossary.md]: How about,

> A cryptographic mechanism that proves that the owner of the private key has approved a transaction or other data (e.g. the hash of a file).

Comment (davecgh) [reply docs/glossary.md]: @jrick's version sounds good to me.

Comment (s-ben) [reply docs/glossary.md]: @davecgh , @jrick , have implemented latest suggestion. Build passing, ready to merge.

**Pull request merged**